### PR TITLE
Add CLI Cmds to use Mamba with Poetry

### PR DESCRIPTION
**Pull request by dwu359**
# Add CLI Cmds to use Mamba with Poetry

**What user problem are we solving?**
Faster installation times, no stupid pyenv version errors, cross-platform packages like pytorch can be better handled by mamba

**What solution does this PR provide?**
backend install cmd creates a conda env named `dlp`, installs all packages in `environment.yml` with mamba, and then installs all packages in `pyproject.toml` with poetry.

backend add, remove cmds add and remove packages mainly from poetry unless specifically specified to install via mamba. 
Unfortunately mamba/conda doesn't provide support by itself to easily update environment.yml automatically, but there's a yellow `IMPORTANT` message at the end followed by instructions on how to update the yml properly for conda packages. Most packages should be installed via poetry though, so most people don't have to worry about this anyways.

backend uninstall cmd added

**Testing Methodology**
How did you test your changes and verify that existing 
functionality is not broken
backend install works
backend add arrow works
backend remove arrow works
backend add keras -m works
backend remove keras -m works

**Any other considerations**

### CODE REVIEW

**Comment by karkir0003** [comment]
@dwu359 wouldn't this PR be made under `dlp-cli` repo and goreleaser pushes the updates automatically?

File: training/pyproject.toml [hunk 1]
```diff
@@ -12,8 +12,6 @@ django = "^4.2.3"
 django-ninja = "^0.22.2"
 scikit-learn = "^1.3.0"
 pandas = "^2.0.3"
-torch = ">=2.0.0, !=2.0.1"
-torchvision = "^0.15"
```
Comment (karkir0003) [inline training/pyproject.toml:16]: What version of `torch` and `torchvision` end up getting installed with this new config? @dwu359

**Comment by karkir0003** [comment]
@dwu359 for the mamba env name, can we make that a constant so that we don't "hardcode" parameters into the main add/remove go scripts we just added since we should force env name in mamba to be `dlp`

Should this install command be rerun even if a returning dev had working installation from before?

**Comment by karkir0003** [comment]
@dwu359 can you put screenshots to show the test results if you can?

**Review by karkir0003 — CHANGES_REQUESTED** [review]
Looks good to me so far. Please address my comments/questions and then re-request review? @dwu359

File: training/pyproject.toml (hunk 1 above)
Comment (dwu359) [reply training/pyproject.toml:16]: On my end, it installed `pytorch 2.0.1` and `torchvision 0.15.2` from the pytorch conda channel. This is specified in environment.yml:
  - pytorch::pytorch
  - pytorch::torchvision

**Comment by dwu359** [comment]
> @dwu359 for the mamba env name, can we make that a constant so that we don't "hardcode" parameters into the main add/remove go scripts we just added since we should force env name in mamba to be `dlp`
> 
> Should this install command be rerun even if a returning dev had working installation from before?

We can, but mamba is also significantly better

**Comment by dwu359** [comment]
@karkir0003 Just added support for ubuntu terminals with the -w flag (runs cmds as if it were in a non-unix terminal)

**Review by karkir0003 — APPROVED** [review]
approved corresponding pr created in `dlp-cli` repo (https://github.com/DSGT-DLP/dlp-cli/pull/3)

**Pull request merged**